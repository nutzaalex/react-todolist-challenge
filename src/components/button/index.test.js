import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import Button from './';

describe('Button', () => {
    it(`matches snapshot for type add`, () => {
        const comp = renderer.create(
            <Button type='add'>Add</Button>
        );

        expect(comp.toJSON()).toMatchSnapshot();
    });
    it(`matches snapshot for type edit`, () => {
        const comp = renderer.create(
            <Button type='edit'>Edit</Button>
        );

        expect(comp.toJSON()).toMatchSnapshot();
    });
    it(`matches snapshot for type default`, () => {
        const comp = renderer.create(
            <Button type='default'>Default</Button>
        );

        expect(comp.toJSON()).toMatchSnapshot();
    });

    it('calls onClick handler', () => {
        const onClickSpy = jest.fn();
        const comp = shallow(
            <Button type="add" onClick={onClickSpy}>
                Add
            </Button>
        );

        expect(onClickSpy).not.toHaveBeenCalled();
        comp.find('button').simulate('click');
        expect(onClickSpy).toHaveBeenCalledTimes(1);
    });
});
