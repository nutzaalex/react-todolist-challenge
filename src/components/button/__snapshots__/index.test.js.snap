// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot for type add 1`] = `
<button
  className="button add"
  onClick={undefined}
>
  Add
</button>
`;

exports[`Button matches snapshot for type default 1`] = `
<button
  className="button default"
  onClick={undefined}
>
  Default
</button>
`;

exports[`Button matches snapshot for type edit 1`] = `
<button
  className="button edit"
  onClick={undefined}
>
  Edit
</button>
`;
